Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2005-05-26
Initial Package Version: 
Origin: 
Upstream Status: 
Description: 
 
diff -Naur bootscripts.orig/Makefile bootscripts/Makefile
--- bootscripts.orig/Makefile	2005-05-26 19:57:15.000000000 +0000
+++ bootscripts/Makefile	2005-05-26 20:02:57.000000000 +0000
@@ -28,6 +28,7 @@
 	install -m ${MODE} lfs/init.d/halt          ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/console       ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/hotplug       ${EXTDIR}/rc.d/init.d/
+	install -m ${MODE} lfs/init.d/paneld        ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/localnet      ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/modules       ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/mountfs       ${EXTDIR}/rc.d/init.d/
@@ -36,32 +37,46 @@
 	install -m ${MODE} lfs/init.d/rc            ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/reboot        ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/sendsignals   ${EXTDIR}/rc.d/init.d/
+	install -m ${MODE} lfs/init.d/sshd          ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/setclock      ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/sysklogd	    ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/swap          ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/sysctl        ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/template      ${EXTDIR}/rc.d/init.d/
 	install -m ${MODE} lfs/init.d/udev          ${EXTDIR}/rc.d/init.d/
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc1.d/K01paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc0.d/K80network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc0.d/K90sysklogd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc0.d/K90sysklogd
 	ln -sf ../init.d/hotplug     ${EXTDIR}/rc.d/rc0.d/S50hotplug
 	ln -sf ../init.d/sendsignals ${EXTDIR}/rc.d/rc0.d/S60sendsignals
 	ln -sf ../init.d/mountfs     ${EXTDIR}/rc.d/rc0.d/S70mountfs
 	ln -sf ../init.d/swap        ${EXTDIR}/rc.d/rc0.d/S80swap
 	ln -sf ../init.d/localnet    ${EXTDIR}/rc.d/rc0.d/S90localnet
 	ln -sf ../init.d/halt        ${EXTDIR}/rc.d/rc0.d/S99halt
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc1.d/K30sshd
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc1.d/S99paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc1.d/K80network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc1.d/K90sysklogd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc1.d/K90sysklogd
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc1.d/K30sshd
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc2.d/S99paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc2.d/K80network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc2.d/K90sysklogd
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc3.d/S10sysklogd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc2.d/K90sysklogd
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc3.d/S30sshd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc3.d/S10sysklogd
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc3.d/S99paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc3.d/S20network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc4.d/S10sysklogd
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc4.d/S30sshd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc4.d/S10sysklogd
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc4.d/S99paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc4.d/S20network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc5.d/S10sysklogd
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc5.d/S30sshd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc5.d/S10sysklogd
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc5.d/S99paneld
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc5.d/S20network
+	ln -sf ../init.d/paneld      ${EXTDIR}/rc.d/rc6.d/K01paneld
+	ln -sf ../init.d/sshd        ${EXTDIR}/rc.d/rc6.d/K30sshd
 	ln -sf ../init.d/network     ${EXTDIR}/rc.d/rc6.d/K80network
-	ln -sf ../init.d/sysklogd   ${EXTDIR}/rc.d/rc6.d/K90sysklogd
+	ln -sf ../init.d/sysklogd    ${EXTDIR}/rc.d/rc6.d/K90sysklogd
 	ln -sf ../init.d/hotplug     ${EXTDIR}/rc.d/rc6.d/S50hotplug
 	ln -sf ../init.d/sendsignals ${EXTDIR}/rc.d/rc6.d/S60sendsignals
 	ln -sf ../init.d/mountfs     ${EXTDIR}/rc.d/rc6.d/S70mountfs
diff -Naur bootscripts.orig/lfs/init.d/checkfs bootscripts/lfs/init.d/checkfs
--- bootscripts.orig/lfs/init.d/checkfs	2005-03-16 22:57:35.000000000 +0000
+++ bootscripts/lfs/init.d/checkfs	2005-05-26 20:02:57.000000000 +0000
@@ -33,11 +33,13 @@
 		if [ -f /fastboot ]; then
 			boot_mesg -n "/fastboot found, will not perform" ${INFO}
 			boot_mesg " file system checks as requested."
+			putlcd "Fast boot requested"
 			echo_ok
 			exit 0
 		fi
 
 		boot_mesg "Mounting root file system in read-only mode..."
+		putlcd "Mounting / in" "read-only mode"
 		mount -n -o remount,ro / >/dev/null
 		evaluate_retval
 
@@ -48,15 +50,16 @@
 			boot_mesg -n " in read-only mode.\n\nAfter you"
 			boot_mesg -n " press Enter, this system will be"
 			boot_mesg -n " halted and powered off."
-			boot_mesg -n "\n\nPress enter to continue..." ${INFO}
 			boot_mesg "" ${NORMAL}
-			read ENTER
+			putlcd "FileSystem Error" "Halting System"
+			sleep 10
 			/etc/rc.d/init.d/halt stop
 		fi
 
 		if [ -f /forcefsck ]; then
 			boot_mesg -n "/forcefsck found, forcing file" ${INFO}
 			boot_mesg " system checks as requested."
+			putlcd "Forcing file" "system check"
 			echo_ok
 			options="-f"
 		else
@@ -64,6 +67,7 @@
 		fi
 
 		boot_mesg "Checking file systems..."
+		putlcd "Checking file"  "systems..."
 		# Note: -a option used to be -p; but this fails e.g.
 		# on fsck.minix
 		fsck ${options} -a -A -C -T 2>/dev/null
@@ -90,9 +94,9 @@
 			boot_mesg -n " errors require this system to be"
 			boot_mesg -n " rebooted.\n\nAfter you press enter,"
 			boot_mesg -n " this system will be rebooted"
-			boot_mesg -n "\n\nPress Enter to continue..." ${INFO}
 			boot_mesg "" ${NORMAL}
-			read ENTER
+			putlcd "Errors corrected" "Rebooting"
+			sleep 5
 			reboot -f
 		fi
 
@@ -107,9 +111,9 @@
 			boot_mesg -n " System Administrator.\n\nAfter you"
 			boot_mesg -n " press Enter, this system will be"
 			boot_mesg -n " halted and powered off."
-			boot_mesg -n "\n\nPress Enter to continue..." ${INFO}
 			boot_mesg "" ${NORMAL}
-			read ENTER
+			putlcd "FileSystem Error" "Halting System"
+			sleep 10
 			/etc/rc.d/init.d/halt stop
 		fi
 
diff -Naur bootscripts.orig/lfs/init.d/cleanfs bootscripts/lfs/init.d/cleanfs
--- bootscripts.orig/lfs/init.d/cleanfs	2005-03-16 22:58:01.000000000 +0000
+++ bootscripts/lfs/init.d/cleanfs	2005-05-26 20:02:57.000000000 +0000
@@ -72,17 +72,21 @@
 case "${1}" in
 	start)
 		boot_mesg -n "Cleaning file systems:" ${INFO}
+		putlcd "Cleaning file" "systems:"
 
 		boot_mesg -n " /tmp" ${NORMAL}
+		putlcd "Cleaning:" "/tmp"
 		cd /tmp &&
 		find . -xdev -mindepth 1 ! -name lost+found \
 			-delete || failed=1
 
 		boot_mesg -n " /var/lock" ${NORMAL}
+		putlcd "Cleaning:" "/var/lock"
 		cd /var/lock &&
 		find . -type f ! -newer /proc -exec rm -f {} \; || failed=1
 
 		boot_mesg " /var/run" ${NORMAL}
+		putlcd "Cleaning:" "/var/run"
 		cd /var/run &&
 		find . ! -type d ! -name utmp ! -newer /proc \
 			-exec rm -f {} \; || failed=1
@@ -97,6 +101,7 @@
 
 		if egrep -qv '^(#|$)' /etc/sysconfig/createfiles 2>/dev/null; then
 			boot_mesg "Creating files and directories..."
+			putlcd "Creating files" "and directories"
 			create_files
 			evaluate_retval
 		fi
diff -Naur bootscripts.orig/lfs/init.d/console bootscripts/lfs/init.d/console
--- bootscripts.orig/lfs/init.d/console	2005-03-16 22:58:21.000000000 +0000
+++ bootscripts/lfs/init.d/console	2005-05-26 20:02:57.000000000 +0000
@@ -25,18 +25,21 @@
 	start)
 		if [ -n "${KEYMAP}" ]; then
 			boot_mesg "Loading keymap: ${KEYMAP}..."
+			putlcd "Loading keymap:" "${KEYMAP}"
 			loadkeys ${KEYMAP} &>/dev/null
 			evaluate_retval
 		fi
 
 		if [ -n "${KEYMAP_CORRECTIONS}" ]; then
 			boot_mesg "Loading keymap corrections: ${KEYMAP_CORRECTIONS}..."
+			putlcd "Loading keymap" "corrections"
 			loadkeys ${KEYMAP_CORRECTIONS} &>/dev/null
 			evaluate_retval
 		fi
 
 		if [ -n "${FONT}" ]; then
 			boot_mesg "Setting screen font to ${FONT}..."
+			putlcd "Setting font" "to $FONT"
 			setfont $FONT &>/dev/null
 			evaluate_retval
 		fi
diff -Naur bootscripts.orig/lfs/init.d/functions bootscripts/lfs/init.d/functions
--- bootscripts.orig/lfs/init.d/functions	2005-03-31 04:10:50.000000000 +0000
+++ bootscripts/lfs/init.d/functions	2005-05-26 20:02:57.000000000 +0000
@@ -263,7 +263,8 @@
 	boot_mesg -n " the LFS book, please be so kind to inform us at"
 	boot_mesg " lfs-dev@linuxfromscratch.org.\n"
 	boot_mesg_flush
-	boot_mesg -n "Press Enter to continue..." ${INFO}
+	putlcd "BootScript Error" "view bootlog"
+	sleep 20
 	boot_mesg "" ${NORMAL}
 	read ENTER
 }
diff -Naur bootscripts.orig/lfs/init.d/hotplug bootscripts/lfs/init.d/hotplug
--- bootscripts.orig/lfs/init.d/hotplug	2005-05-23 19:13:29.000000000 +0000
+++ bootscripts/lfs/init.d/hotplug	2005-05-26 20:02:57.000000000 +0000
@@ -18,6 +18,7 @@
 case "${1}" in
 	start|restart)
 		boot_mesg "Loading modules for system devices..."
+		putlcd "Loading modules" "for devices"
 		for rc in /etc/hotplug/*.rc
 		do
 			${rc} ${1}
diff -Naur bootscripts.orig/lfs/init.d/localnet bootscripts/lfs/init.d/localnet
--- bootscripts.orig/lfs/init.d/localnet	2005-05-23 19:13:29.000000000 +0000
+++ bootscripts/lfs/init.d/localnet	2005-05-26 20:04:44.000000000 +0000
@@ -19,17 +19,20 @@
 case "${1}" in
 	start)
 		boot_mesg "Bringing up the loopback interface..."
+		putlcd "Loopback:" "Up"
 		ip addr add 127.0.0.1/8 label lo dev lo
 		ip link set lo up
 		evaluate_retval
 
 		boot_mesg "Setting hostname to ${HOSTNAME}..."
+		putlcd "Hostname:" "${HOSTNAME}"
 		hostname ${HOSTNAME}
 		evaluate_retval
 		;;
 
 	stop)
 		boot_mesg "Bringing down the loopback interface..."
+		putlcd "Loopback:" "Down"
 		ip link set lo down
 		evaluate_retval
 		;;
diff -Naur bootscripts.orig/lfs/init.d/modules bootscripts/lfs/init.d/modules
--- bootscripts.orig/lfs/init.d/modules	2005-03-16 23:03:11.000000000 +0000
+++ bootscripts/lfs/init.d/modules	2005-05-26 20:02:57.000000000 +0000
@@ -60,6 +60,7 @@
 					# otherwise take note.
 					if [ ${?} -eq 0 ]; then
 						boot_mesg -n " ${module}" ${NORMAL}
+						putlcd "Module:" "${module}"
 					else
 						failedmod="${failedmod} ${module}"
 					fi
diff -Naur bootscripts.orig/lfs/init.d/mountfs bootscripts/lfs/init.d/mountfs
--- bootscripts.orig/lfs/init.d/mountfs	2005-03-12 01:31:41.000000000 +0000
+++ bootscripts/lfs/init.d/mountfs	2005-05-26 20:02:57.000000000 +0000
@@ -18,6 +18,7 @@
 case "${1}" in
 	start)
 		boot_mesg "Remounting root file system in read-write mode..."
+		putlcd "Remounting:" "/"
 		mount -n -o remount,rw / &>/dev/null
 		evaluate_retval
 
@@ -25,6 +26,7 @@
 		rm -f /fastboot /forcefsck
 
 		boot_mesg "Recording existing mounts in /etc/mtab..."
+		putlcd "Recording mounts" "in /etc/mtab"
 		> /etc/mtab
 		mount -f / || failed=1
 		mount -f /proc || failed=1
@@ -35,12 +37,14 @@
 		# This will mount all filesystems that do not have _netdev in
 		# their option list.  _netdev denotes a network filesystem.
 		boot_mesg "Mounting remaining file systems..."
+		putlcd "Mounting:" "file systems"
 		mount -a -O no_netdev &>/dev/null
 		evaluate_retval
 		;;
 
 	stop)
 		boot_mesg "Unmounting all other currently mounted file systems..."
+		putlcd "Unmounting:" "file systems"
 		umount -a -d -r &>/dev/null
 		evaluate_retval
 		;;
diff -Naur bootscripts.orig/lfs/init.d/paneld bootscripts/lfs/init.d/paneld
--- bootscripts.orig/lfs/init.d/paneld	1970-01-01 00:00:00.000000000 +0000
+++ bootscripts/lfs/init.d/paneld	2005-05-26 20:02:57.000000000 +0000
@@ -0,0 +1,63 @@
+#!/bin/sh
+########################################################################
+# Begin $rc_base/init.d/paneld
+#
+# Description : Show Messages on the LCD Panel
+#
+# Author      : Jim Gifford - jim@linuxfromscratch.org
+#
+# Version     : 00.00
+#
+# Notes       :
+#
+########################################################################
+
+. /etc/sysconfig/rc
+. $rc_functions
+
+case "$1" in
+	start)
+		boot_mesg "Setting up LCD Panel"
+		ip addr show dev eth0 > /dev/null 2>&1
+		if [ "$?" = "0" ]; then
+			IFACE1_IP="`ip addr show dev eth0 | grep inet | cut -d / -f 1 | cut -d t  -f 2 | cut -d' ' -f 2 `"
+			IFACE1="$IFACE1_IP"
+		fi
+		ip addr show dev eth1 > /dev/null 2>&1
+		if [ "$?" = "0" ]; then
+			IFACE2_IP="`ip addr show dev eth1 | grep inet | cut -d / -f 1 | cut -d t  -f 2 | cut -d' ' -f 2 `"
+			IFACE2="$IFACE2_IP"
+		fi
+		COMMON="Hold S & E Menu"
+		if [ "$IFACE2" = "" ]; then
+			loadproc paneld -d "Lan1 $IFACE1" "$COMMON"
+		else
+			loadproc paneld -d "Lan1 $IFACE1" "Lan2 $IFACE2" "$COMMON"
+		fi
+
+	;;
+
+        stop)
+                boot_mesg "Clearing LCD Panel..."
+                putlcd "Clearing" "LCD Panel"
+                killproc paneld
+                ;;
+
+        restart)
+                ${0} stop
+                sleep 1
+                ${0} start
+                ;;
+
+        status)
+                statusproc paneld
+                ;;
+
+	*)
+		echo "Usage: {start|stop|restart|status}"
+		exit 1
+		;;
+esac
+
+# End $rc_base/init.d/paneld
+
diff -Naur bootscripts.orig/lfs/init.d/rc bootscripts/lfs/init.d/rc
--- bootscripts.orig/lfs/init.d/rc	2005-03-16 23:04:42.000000000 +0000
+++ bootscripts/lfs/init.d/rc	2005-05-26 20:02:57.000000000 +0000
@@ -33,6 +33,7 @@
 
 if [ ! -d ${rc_base}/rc${runlevel}.d ]; then
 	boot_mesg "${rc_base}/rc${runlevel}.d does not exist." ${WARNING}
+	putlcd "${rc_base}/rc${runlevel}.d" "does not exist."
 	boot_mesg_flush
 	exit 1
 fi
@@ -56,6 +57,8 @@
 				boot_mesg -n " not started in the previous"
 				boot_mesg -n " runlevel (${previous})."
 				boot_mesg "" ${NORMAL}
+				putlcd "BootScript Error" "view bootlog"
+				sleep 20
 				boot_mesg_flush
 				continue
 			fi
@@ -92,7 +95,7 @@
 	esac
 	error_value=${?}
 
-	if [ "${error_value}" != "0" ]; then
+	if [ "${error_value}" != "0" ];	then
 		print_error_msg
 	fi
 done
diff -Naur bootscripts.orig/lfs/init.d/reboot bootscripts/lfs/init.d/reboot
--- bootscripts.orig/lfs/init.d/reboot	2005-03-12 01:31:41.000000000 +0000
+++ bootscripts/lfs/init.d/reboot	2005-05-26 20:02:57.000000000 +0000
@@ -17,7 +17,6 @@
 
 case "${1}" in
 	stop)
-		boot_mesg "Restarting system..."
 		reboot -d -f -i
 		;;
 
diff -Naur bootscripts.orig/lfs/init.d/sendsignals bootscripts/lfs/init.d/sendsignals
--- bootscripts.orig/lfs/init.d/sendsignals	2005-03-16 23:05:03.000000000 +0000
+++ bootscripts/lfs/init.d/sendsignals	2005-05-26 20:02:57.000000000 +0000
@@ -18,6 +18,7 @@
 case "${1}" in
 	stop)
 		boot_mesg "Sending all processes the TERM signal..."
+		putlcd "Sending:" "TERM signal"
 		killall5 -15
 		error_value=${?}
 
@@ -30,6 +31,7 @@
 		fi
 
 		boot_mesg "Sending all processes the KILL signal..."
+		putlcd "Sending:" "KILL signal"
 		killall5 -9
 		error_value=${?}
 
diff -Naur bootscripts.orig/lfs/init.d/setclock bootscripts/lfs/init.d/setclock
--- bootscripts.orig/lfs/init.d/setclock	2005-03-12 01:31:41.000000000 +0000
+++ bootscripts/lfs/init.d/setclock	2005-05-26 20:02:57.000000000 +0000
@@ -32,12 +32,14 @@
 case ${1} in
 	start)
 		boot_mesg "Setting system clock..."
+		putlcd "Setting:" "system clock"
 		hwclock --hctosys ${CLOCKPARAMS} &>/dev/null
 		evaluate_retval
 		;;
 
 	stop)
 		boot_mesg "Setting hardware clock..."
+		putlcd "Setting:" "hardware clock"
 		hwclock --systohc ${CLOCKPARAMS} &>/dev/null
 		evaluate_retval
 		;;
diff -Naur bootscripts.orig/lfs/init.d/sshd bootscripts/lfs/init.d/sshd
--- bootscripts.orig/lfs/init.d/sshd	1970-01-01 00:00:00.000000000 +0000
+++ bootscripts/lfs/init.d/sshd	2005-05-26 20:02:57.000000000 +0000
@@ -0,0 +1,52 @@
+#!/bin/sh
+########################################################################
+# Begin $rc_base/init.d/sshd
+#
+# Description : SSH Server
+#
+# Author      :
+#
+# Version     : 00.01
+#
+# Notes       :
+#
+########################################################################
+
+. /etc/sysconfig/rc
+. $rc_functions
+
+case "$1" in
+	start)
+		boot_mesg "Starting SSH Server..."
+		putlcd "Starting:" "SSH Server"
+		loadproc /usr/sbin/sshd
+		;;
+
+	stop)
+		boot_mesg "Stopping SSH Server..."
+		putlcd "Stopping:" "SSH Server"
+		killproc /usr/sbin/sshd
+		;;
+
+	reload)
+		boot_mesg "Reloading SSH Server..."
+		reloadproc /usr/sbin/sshd
+		;;
+
+	restart)
+		$0 stop
+		sleep 1
+		$0 start
+		;;
+
+	status)
+		statusproc /usr/sbin/sshd
+		;;
+
+	*)
+		echo "Usage: $0 {start|stop|reload|restart|status}"
+		exit 1
+		;;
+esac
+
+# End $rc_base/init.d/sshd
diff -Naur bootscripts.orig/lfs/init.d/swap bootscripts/lfs/init.d/swap
--- bootscripts.orig/lfs/init.d/swap	2005-03-12 01:31:41.000000000 +0000
+++ bootscripts/lfs/init.d/swap	2005-05-26 20:02:57.000000000 +0000
@@ -18,12 +18,14 @@
 case "${1}" in
 	start)
 		boot_mesg "Activating all swap files/partitions..."
+		putlcd "Activating:" "swap"
 		swapon -a
 		evaluate_retval
 		;;
 
 	stop)
 		boot_mesg "Deactivating all swap files/partitions..."
+		putlcd "Deactivating:" "swap"
 		swapoff -a
 		evaluate_retval
 		;;
diff -Naur bootscripts.orig/lfs/init.d/sysctl bootscripts/lfs/init.d/sysctl
--- bootscripts.orig/lfs/init.d/sysctl	2005-03-16 23:05:21.000000000 +0000
+++ bootscripts/lfs/init.d/sysctl	2005-05-26 20:02:57.000000000 +0000
@@ -21,6 +21,7 @@
 	start)
 		if [ -f "/etc/sysctl.conf" ]; then
 			boot_mesg "Setting kernel runtime parameters..."
+			putlcd "Utilizing:" "sysctl"
 			sysctl -q -p
 			evaluate_retval
 		fi
diff -Naur bootscripts.orig/lfs/init.d/sysklogd bootscripts/lfs/init.d/sysklogd
--- bootscripts.orig/lfs/init.d/sysklogd	2005-03-31 04:10:50.000000000 +0000
+++ bootscripts/lfs/init.d/sysklogd	2005-05-26 20:02:57.000000000 +0000
@@ -18,17 +18,21 @@
 case "${1}" in
 	start)
 		boot_mesg "Starting system log daemon..."
+		putlcd "Starting:" "system log"
 		loadproc syslogd -m 0
 
 		boot_mesg "Starting kernel log daemon..."
+		putlcd "Starting:" "kernel log"
 		loadproc klogd
 		;;
 
 	stop)
 		boot_mesg "Stopping kernel log daemon..."
+		putlcd "Stopping:" "system log"
 		killproc klogd
 
 		boot_mesg "Stopping system log daemon..."
+		putlcd "Stopping:" "kernel log"
 		killproc syslogd
 		;;
 
diff -Naur bootscripts.orig/lfs/init.d/udev bootscripts/lfs/init.d/udev
--- bootscripts.orig/lfs/init.d/udev	2005-03-16 23:05:42.000000000 +0000
+++ bootscripts/lfs/init.d/udev	2005-05-26 20:02:57.000000000 +0000
@@ -39,7 +39,8 @@
 			boot_mesg -n " will be halted and powered off."
 			boot_mesg -n "\n\nPress Enter to continue..." ${INFO}
 			boot_mesg "" ${NORMAL}
-			read ENTER
+			putlcd "BootScript Error" "view bootlog"
+			sleep 20
 			/etc/rc.d/init.d/halt stop
 		fi
 
@@ -57,7 +58,8 @@
 			boot_mesg -n " will be halted and powered off."
 			boot_mesg -n "\n\nPress Enter to continue..." ${INFO}
 			boot_mesg "" ${NORMAL}
-			read ENTER
+			putlcd "BootScript Error" "view bootlog"
+			sleep 20
 			/etc/rc.d/init.d/halt stop
 		fi
 
diff -Naur bootscripts.orig/lfs/sysconfig/network-devices/ifdown bootscripts/lfs/sysconfig/network-devices/ifdown
--- bootscripts.orig/lfs/sysconfig/network-devices/ifdown	2005-03-31 17:12:11.000000000 +0000
+++ bootscripts/lfs/sysconfig/network-devices/ifdown	2005-05-26 20:02:57.000000000 +0000
@@ -69,6 +69,7 @@
 				IFCONFIG=${file} ${network_devices}/services/${SERVICE} ${1} down
 			else
 				boot_mesg "Interface ${1} doesn't exist." ${WARNING}
+				putlcd "Interface ${1}" "doesn't exist."
 				echo_warning
 			fi
 		else
@@ -86,6 +87,7 @@
 	if [ -n "${link_status}" ]; then
 		if echo "${link_status}" | grep -q UP; then
 			boot_mesg "Bringing down the ${1} interface..."
+			putlcd "Bringing down" "${1} interface"
 			ip link set ${1} down
 			evaluate_retval
 		fi
diff -Naur bootscripts.orig/lfs/sysconfig/network-devices/ifup bootscripts/lfs/sysconfig/network-devices/ifup
--- bootscripts.orig/lfs/sysconfig/network-devices/ifup	2005-03-31 17:22:48.000000000 +0000
+++ bootscripts/lfs/sysconfig/network-devices/ifup	2005-05-26 20:02:57.000000000 +0000
@@ -31,6 +31,7 @@
 fi
 
 boot_mesg "Bringing up the ${1} interface..."
+putlcd "Bringing up" "${1} interface"
 boot_mesg_flush
 
 # Process each configruation file
@@ -71,6 +72,7 @@
 					fi
 				else
 					boot_mesg "Interface ${1} doesn't exist." ${WARNING}
+					putlcd "Interface ${1}" "doesn't exist."
 					echo_warning
 					continue
 				fi
diff -Naur bootscripts.orig/lfs/sysconfig/network-devices/services/ipv4-static bootscripts/lfs/sysconfig/network-devices/services/ipv4-static
--- bootscripts.orig/lfs/sysconfig/network-devices/services/ipv4-static	2005-03-31 06:15:31.000000000 +0000
+++ bootscripts/lfs/sysconfig/network-devices/services/ipv4-static	2005-05-26 20:02:57.000000000 +0000
@@ -50,6 +50,7 @@
 case "${2}" in
 	up)
 		boot_mesg "Adding IPv4 address ${IP} to the ${1} interface..."
+		putlcd "Setting" "${1} IP - ${IP}"
 		ip addr add ${args} dev ${1}
 		evaluate_retval
 	
@@ -59,6 +60,7 @@
 				echo_warning
 			else
 				boot_mesg "Setting up default gateway..."
+				putlcd "Setting up" "default gateway"
 				ip route add default via ${GATEWAY} dev ${1}
 				evaluate_retval
 			 fi
@@ -68,11 +70,13 @@
 	down)
 		if [ -n "${GATEWAY}" ];	then
 			boot_mesg "Removing default gateway..."
+			putlcd "Removing" "default gateway"
 			ip route del default
 			evaluate_retval
 		fi
 	
 		boot_mesg "Removing IPv4 address ${IP} from the ${1} interface..."
+		putlcd "Unsetting" "${1} IP - ${IP}"
 		ip addr del ${args} dev ${1}
 		evaluate_retval
 	;;
diff -Naur bootscripts.orig/lfs/sysconfig/network-devices/services/ipv4-static-route bootscripts/lfs/sysconfig/network-devices/services/ipv4-static-route
--- bootscripts.orig/lfs/sysconfig/network-devices/services/ipv4-static-route	2005-03-31 06:15:43.000000000 +0000
+++ bootscripts/lfs/sysconfig/network-devices/services/ipv4-static-route	2005-05-26 20:02:57.000000000 +0000
@@ -74,12 +74,14 @@
 case "${2}" in
 	up)
 		boot_mesg "Adding '${desc}' route to the ${1} interface..."
+		putlcd "Adding route" "${desc}"
 		ip route add ${args} dev ${1}
 		evaluate_retval
 	;;
 	
 	down)
 		boot_mesg "Removing '${desc}' route from the ${1} interface..."
+		putlcd "Removing route" "${desc}"
 		ip route del ${args} dev ${1}
 		evaluate_retval
 	;;
